packages/magic-link/index.js

```diff
@@ -83,18 +83,14 @@ function MagicLink(options) {
  *
  * @param {object} options
  * @param {string} options.email - The email to send magic link to
- * @param {string} options.labels - The labels for member
+ * @param {string} options.payload - The payload for token
  * @param {object} options.subject - The subject to associate with the magic link (user id, or email)
  * @param {string=} [options.type='signin'] - The type to be passed to the url and content generator functions
  * @returns {Promise<{token: JSONWebToken, info: SentMessageInfo}>}
  */
 MagicLink.prototype.sendMagicLink = async function sendMagicLink(options) {
-    const labels = options.labels;
-    const payload = {};
-    if (labels) {
-        payload.labels = labels;
-    }
-    const token = jwt.sign(payload, this.secret, {
+    const payload = options.payload || {};
+    const token = jwt.sign({payload}, this.secret, {
```

Comment:
This should just be `jwt.sign(payload, this.secret, {` I think?

Reply:
@allouis This was needed as jwt passes keys in the payload object as-is in the `claims` on decoding with `jwt.verify`, so a flat structure here will mean individually reading them back in decode method. The wrapping payload data this way allows reading of full payload from `claims` directly by simply doing `claims.payload` after decoding token - https://github.com/TryGhost/Members/pull/124/files#diff-7a8dcd1ee8e23bda9b77c795706c6046R140

Reply:
I think you could probably just rename that to getClaims OR just return the claims object tbh as it will have all the expected data.

If you don't wanna do that then I think we should use a key like `p` or `pld` so that we keep the tokens small as possible

Reply:
> just return the claims object tbh as it will have all the expected data.

Yeah this ones probably the best in that case. 👍

Reply:
@allouis Updated the PR with changes including flat structure for payload directly. Lmk if it looks good to merge, thanks 😄